tests/Unit/Cli/CampaignValidatorTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Tests\Unit\Cli;
+
+use WMDE\Fundraising\Frontend\BucketTesting\Bucket;
+use WMDE\Fundraising\Frontend\BucketTesting\Campaign;
+use WMDE\Fundraising\Frontend\BucketTesting\CampaignCollection;
+use WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\CampaignValidator;
+use WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\ValidationErrorLogger;
+use WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\Rule\DefaultBucketRule;
+use WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\Rule\StartAndEndTimeRule;
+use WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\Rule\UniqueBucketRule;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\FakeBucket;
+
+/**
+ * @covers \WMDE\Fundraising\Frontend\Cli\CampaignConfigValidation\CampaignValidator
+ */
+class CampaignValidatorTest extends \PHPUnit\Framework\TestCase {
+
+	public function testWhenValidCampaignIsSupplied_validationPasses(): void {
+		$errorLogger = new ValidationErrorLogger();
+		$bucket = FakeBucket::createBucket();
+		$campaignCollection = new CampaignCollection( $bucket->getCampaign() );
+
+		$validator = new CampaignValidator( $campaignCollection, $errorLogger );
+		$this->assertTrue( $validator->isPassing() );
+	}
+
+	public function testWhenNoDefaultBucketIsSupplied_validationFails(): void {
+		$errorLogger = new ValidationErrorLogger();
+		$bucket = FakeBucket::createBucket( 'test', false );
+
+		$rule = new DefaultBucketRule();
+		$rule->validate( $bucket->getCampaign(), $errorLogger );
+		$this->assertEquals( ['test_campaign: Must have a valid default bucket.'], $errorLogger->getErrors() );
+	}
+
+	public function testWhenCampaignEndTimeIsLowerThanOrEqualToStartTime_validationFails(): void {
+		$errorLogger = new ValidationErrorLogger();
+		$campaign = new Campaign(
+			'test_campaign',
+			'test_campaign',
+			new \DateTime( '2099-01-01' ),
+			new \DateTime( '2099-01-01' ),
+			true
+		);
+		$rule = new StartAndEndTimeRule();
+		$rule->validate( $campaign, $errorLogger );
+		$this->assertEquals( ['test_campaign: Start date must be before end date.'], $errorLogger->getErrors() );
+	}
+
+	public function testWhenCampaignBucketsAreNotUnique_validationFails(): void {
+		$errorLogger = new ValidationErrorLogger();
+		$bucket = FakeBucket::createBucket( 'test', true );
+		$bucket->getCampaign()->addBucket( new Bucket( 'test', $bucket->getCampaign(), false ) );
+
+		$rule = new UniqueBucketRule();
+		$rule->validate( $bucket->getCampaign(), $errorLogger );
+		$this->assertEquals( ['test_campaign: Duplicate bucket test'], $errorLogger->getErrors() );
+	}
+}
```

Comment:
Not sure if the campaigns and buckets should be initialized with `new` statements to make it more clear what's actually in them.